tools/smart_locator_finder.py

```
@@ -474,6 +474,360 @@ async def _find_element_by_description(page, description: str) -> Optional[str]:
+        logger.debug(f"   ⚠️ No structured table_cell_info provided for: {description}")
+        return None
+    
+    # Extract structured info
+    table_heading = table_cell_info.get('table_heading')
+    table_index = table_cell_info.get('table_index', 1)
+    row = table_cell_info.get('row')
+    column = table_cell_info.get('column')
+    
+    # Validate required fields
+    if row is None or column is None:
+        logger.warning(f"   ⚠️ Missing row ({row}) or column ({column}) in table_cell_info")
```

Ruff (0.14.8)

Logging statement uses f-string

(G004)